src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js

```diff
@@ -155,15 +149,42 @@
         // returns map of sobject name => list of fields
         var allFieldsBySObject = this.groupFieldsBySObject(everyField);
 
-        Object.keys(allFieldsBySObject).forEach(function(sObjectName) {
+        var fieldsBySObjectInOrder = {};
+
+        const OPPORTUNITY_SOBJECT_NAME = "Opportunity";
+        const PAYMENT_SOBJECT_NAME = "Payment";
+        // Make sure Opportunity is always the first sObject to be shown
+        if (OPPORTUNITY_SOBJECT_NAME in allFieldsBySObject) {
```

Comment:
I'm not actually positive if this is cleaner, but for the sake of sharing a thought - you could just create an ordered list of the keys without actually copying to a new map, because you can iterate over the ordered keys but index the original map since the values should be the same, just re-ordred (the below code is just example, not great use of var over let and variable naming etc) - 

var sObjectKeys = Object.keys(allFieldsBySObject);
var orderedKeys = [];
for(var i=0; i<sObjectKeys.length; i++) {
   var key = sObjectKeys[i];
   if(key === opportunitySObjectName || key === paymentSObjectName) {
      orderedKeys.unshift(key);  
   } else {
      orderedKeys.push(key);
   }
}

// payment and opp are first and second but in unpredictable order
// swap them if payment is first
if(orderedKeys.indexOf(opportunitySObjectName) > 0) {
  var firstElement = orderedKeys[0];
  orderedKeys.copyWithin(0,1,2);
  orderedKeys[1] = firstElement;
}
orderedKeys.forEach(function(sObjectName) {
  ...
});